### PR TITLE
Enable the deterministic random number generator on GPU

### PR DESCRIPTION
**Pull request by sjsprecious**
This PR enables the deterministic random number on GPU by adding the OpenACC directives.

The GPU-enabled deterministic random number is later used in the droplet subroutine for re-injection of fallout droplets.

The results indicate that:
- the 1st-order droplet diagnostic is consistent between CPU and GPU runs.
- the 0th-order droplet diagnostic differs at a small magnitude of relative error.

**CPU runs: intel vs. openacc**
image

CPU run (intel) vs. MPI+GPU (nvhpc)
image

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -296,7 +296,11 @@ OBJS_M = $(addsuffix .o, $(basename $(SRC_M)))
 OBJS_OPENACC = $(addsuffix .o, $(basename $(SRC_NVTX)))
 
 FFLAGS  = $(OPTS)
-FFLAGS_M  = $(OPTS_M)
+ifeq (${USE_OPENACC_L},true)
+FFLAGS_M = $(OPTS_M) -L/glade/u/apps/opt/nvhpc/22.2/Linux_x86_64/22.2/cuda/11.6/lib64 -lcudart
```
Comment (johnmauff) [inline src/Makefile]: @sjsprecious So this change to the link line was due to adding a CUDA-based memory measurement right? It probably will not be used very frequently.  Maybe we should switch off this link option by default?

**Review by sherimickelson — APPROVED** [review]
@sjsprecious Thanks for adding this in.  It'll really help with verifying our future modifications.

File: src/Makefile (hunk 1 above)
Comment (johnmauff) [reply src/Makefile]: Thanks @sjsprecious, this PR now looks good.

Comment (sjsprecious) [reply src/Makefile]: Thanks @johnmauff. I have commented out the lib path as you suggested.

**Review by johnmauff — APPROVED** [review]

**Pull request merged**